pxtcompiler/emitter/backjs.ts

```diff
@@ -221,7 +221,7 @@ s.pc = -1;
         }
         writeRaw(`
 while (true) {
-if (yieldSteps-- < 0 && maybeYield(s, step, r0)) return null;
+if (yieldSteps-- < 0 && maybeYield(s, step, r0) || runtime.id !== pxsim.runtime.id) return null;
```

Comment:
can you do this? I think it would be slightly faster.

```suggestion
if (yieldSteps-- < 0 && maybeYield(s, step, r0) || runtime !== pxsim.runtime) return null;
```